File: app/src/main/java/com/leilei/guoshujinfu/mylearning/util/AppCache.java
package com.leilei.guoshujinfu.mylearning.util;

import android.content.Context;
import android.content.SharedPreferences;
import android.graphics.drawable.Drawable;

import com.leilei.guoshujinfu.mylearning.R;

/**
 * Created by Administrator on 2017/8/3.
 */

public final class AppCache {
    private static Context mContext;
    private static String mFileName;
    private SharedPreferences mSharedPreferences;
    private SharedPreferences.Editor mEditor;

    private AppCache () {
        if(mContext == null) {
            throw new RuntimeException("must invoke init() before new an object");
        }
        mSharedPreferences = mContext.getSharedPreferences(mFileName,Context.MODE_PRIVATE);
        mEditor = mSharedPreferences.edit();
    }
    public static void init(Context context) {
        mContext = context;
    }
    public static void setFileName(String fileName) {
        mFileName = fileName;
    }
    public static AppCache getAppCache() {
        return  CacheHolder.mCache;
    }
    public AppCache putString(String key, String defaultValue) {
        mEditor.putString(key, defaultValue);
        return this;
    }
    public AppCache getString(String key, String defaultValue) {
        mSharedPreferences.getString(key, defaultValue);
        return this;
    }
    public void commit() {
        mEditor.commit();
    }
    public AppCache remove(String key) {
        mEditor.remove(key);
        return this;
    }


    private static class CacheHolder {
        private static final AppCache mCache = new AppCache();
    }

    public static Drawable getAvatar(Context context){
        return context.getResources().getDrawable(R.mipmap.main_img_tab_profile_blue, null);
    }
}

File: yueting/src/main/kotlin/com/aqrlei/graduation/yueting/util/AppCache.kt
package com.aqrlei.graduation.yueting.util

import android.content.Context
import android.content.SharedPreferences

/**
 * @Author: AqrLei
 * @CreateTime: Date: 2017/9/12 Time: 14:54
 */
enum class AppCache {
    APPCACHE;

    companion object {
        private lateinit var mContext: Context
        private var mFileName: String? = ""
        private lateinit var mShared: SharedPreferences
        private lateinit var mEditor: SharedPreferences.Editor
        fun init(context: Context, name: String?) {
            mContext = context.applicationContext
            mFileName = name
            mShared = mContext.getSharedPreferences(mFileName, Context.MODE_PRIVATE)
            mEditor = mShared.edit()
        }
    }

    fun putString(key: String, value: String): AppCache {
        mEditor.putString(key, value)
        return this
    }

    fun putFloat(key: String, value: Float): AppCache {
        mEditor.putFloat(key, value)
        return this
    }

    fun putLong(key: String, value: Long): AppCache {
        mEditor.putLong(key, value)
        return this
    }

    fun putInt(key: String, value: Int): AppCache {
        mEditor.putInt(key, value)
        return this
    }

    fun putBoolean(key: String, value: Boolean): AppCache {
        mEditor.putBoolean(key, value)
        return this
    }

    fun commit() {
        mEditor.commit()
        mEditor.clear()
    }

    fun remove(key: String): AppCache {
        mEditor.remove(key)
        return this
    }

    fun removeAll(): AppCache {
        mEditor.clear()
        return this
    }

    fun getString(key: String, defValue: String) = mShared.getString(key, defValue)
    fun getFloat(key: String, defValue: Float) = mShared.getFloat(key, defValue)
    fun getLong(key: String, defValue: Long) = mShared.getLong(key, defValue)
    fun getInt(key: String, defValue: Int) = mShared.getInt(key, defValue)
    fun getBoolean(key: String, defValue: Boolean) = mShared.getBoolean(key, defValue)

}